components/kserve/kserve.go

```diff
@@ -131,6 +132,12 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client,
 				return err
 			}
 		}
+		extraParamsMap := map[string]string{
+			"nim-state": string(k.NIM.ManagementState),
```

Comment:
My understanding is that here we are updating the `params.env` only in the `KServe` is `Managed`, if not, the value in the  `params.env` is untouched. However in the `ModelMesh` component reconciliation, the `params.env` value is updated regardless of the status of the `KServe`  component. 

It is probably not an issue but wonder if the behavior should be made consistent.

Reply:
The assumption is that NIM depends on Kserve. If Kserve is not managed, NIM shouldn't be either. If Kserve is not managed, is the Kserve component's reconciler running? How would you suggest making the behaviour consistent?

Reply:
So if the Kserve's component has some cleanup logic in place, can we use it to set `nim-state` to `removed` if `kserve` is removed? That way, we can make both behaviours consistent.

Reply:
Yes. it is the same `ReconcileComponent`  metho, there is a check `enabled := k.GetManagementState() == operatorv1.Managed`

Reply:
Is component cleanup also based on kustomization? If so, isn't it removing the configmap?

Reply:
Is the ConfigMap being deleted when Kserve's component is removed?

Reply:
Yes it is, so yes, it is not an issue, my point was mainly about consistency in the code.  
But I'm fine to leave it as it is.

Reply:
can we update line120-124 to  this:
```
if !enabled {
		if err := deploy.ApplyParams(DependentPath, nil, map[string]string{"nim-state": "removed"}); err != nil {
			return fmt.Errorf("failed to update NIM flag to removed : %w", err)
		}
		if err := k.removeServerlessFeatures(ctx, cli, owner, dscispec); err != nil {
			return err
		}
	} else {
...
```

Reply:
@zdtsw ✅  Done